index.js

```diff
@@ -41,20 +41,26 @@ export class Dagula {
    */
   async * get (cid, options = {}) {
     cid = typeof cid === 'string' ? CID.parse(cid) : cid
-    const order = options.order ?? 'dfs'
     log('getting DAG %s', cid)
-    let cids = Array.isArray(cid) ? cid : [cid]
+
+    const order = options.order ?? 'dfs'
+
+    // fn to track which links to follow next
+    const search = order === 'dfs' ? depthFirst() : breadthFirst()
+
+    // fn to normalize extracting links from different block types
     const getLinks = blockLinks(options.filter)
 
+    let cids = search(Array.isArray(cid) ? cid : [cid])
+
     /** @type {AbortController[]} */
     let aborters = []
     const { signal } = options
     signal?.addEventListener('abort', () => aborters.forEach(a => a.abort(signal.reason)))
 
     while (cids.length > 0) {
       log('fetching %d CIDs', cids.length)
-      const parallelFn = order === 'dfs' ? parallelMap : transform
-      const fetchBlocks = parallelFn(cids.length, async cid => {
+      const fetchBlocks = transform(cids.length, async cid => {
```

Comment:
we don't need to worry about ordering here for `dfs`. the search function will only return a single cid to resolve when it's dfs.